### PR TITLE
fix(code-snippet): improve copy button overlay presentation

### LINKED ISSUE #11872: [code-snippet]: Text can be obscured by the copy button
### Description

The copy button in the code snippet can overlay the text in the component making the text difficult to read and the copy button not immediately identifiable. While in some cases the code can scroll horizontally, edge cases exist where that's not possible.

Screenshot 2024-06-17 at 9 13 55 AM

### Component(s) impacted

bx/cds-code-snippet

### Browser

Firefox

### Carbon for IBM.com version

v1 & v2

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

AEM

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://web-components-v1.carbondesignsystem.com/iframe.html?id=components-code-snippet--multi-line&viewMode=story

### Steps to reproduce the issue (if applicable)

_No response_

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by jkaeser**
### Related Ticket(s)

Closes #11872

### Description

The `<bx-code-snippet>` copy button is positioned absolutely, creating scenarios where it can overlay text in the component body. Giving it a greater `z-index` value allows it to sit atop the text rather than gets mashed into the text.

### Changelog

**Changed**

- Prevents `<bx-code-snippet>` copy button from obscuring component text and becoming difficult to identify in certain scenarios.

### CODE REVIEW

File: packages/carbon-web-components/src/components/code-snippet/code-snippet.scss [hunk 1]
```diff
@@ -93,12 +93,17 @@ $css--plex: true !default;
...
+    z-index: 1;
+  }
+
   .#{$prefix}--snippet-container {
     display: block;
     overflow: hidden;
     position: relative;
     height: auto;
     padding-bottom: 0;
+    padding-right: $carbon--spacing-07;
```
Comment (m4olivei) [inline packages/carbon-web-components/src/components/code-snippet/code-snippet.scss:106]: Wondering if this should be `1.5rem`, such that the code buts right up against the copy button, which is how the single line code snippet is currently:

image

Here is how it is with this change:

image

**Review by m4olivei — APPROVED** [review]
Code changes look good, fixes the issue. Wondering about the spacing, maybe we can get a deign opinion here.

File: packages/carbon-web-components/src/components/code-snippet/code-snippet.scss (hunk 1 above)
Comment (m4olivei) [reply packages/carbon-web-components/src/components/code-snippet/code-snippet.scss:106]: Although I could see the argument going for the 0.5rem spacing all around the copy button 🤷. Either way, I'm good with it 😀.

Comment (jkaeser) [reply packages/carbon-web-components/src/components/code-snippet/code-snippet.scss]: I like that suggestion! Use as little space as possible while still allowing the text to clear the button once scrolled all the way to the right.

**Comment by RichKummer** [comment]
Hey @jkaeser , it looks like the web components version in this PR is missing the overflow gradient which fades out the overlapping text before getting to the copy button. Could this be added here to make the transition clearer?

**This PR**
Screenshot 2024-06-24 at 10 18 20 AM

**React**
Screenshot 2024-06-24 at 10 17 49 AM

Screenshot 2024-06-24 at 10 18 08 AM

**Comment by RichKummer** [comment]
Hey @jkaeser , you might still be working on updates for this – I'm seeing the gradient for both `single` and `multi-line`, but I did notice the padding for the `inline` version is off with this PR:
Screenshot 2024-06-24 at 2 39 35 PM

**Comment by jkaeser** [comment]
> Hey @jkaeser , you might still be working on updates for this – I'm seeing the gradient for both `single` and `multi-line`, but I did notice the padding for the `inline` version is off with this PR

Yep, still working on that one!

**Comment by jkaeser** [comment]
@RichKummer Fixed the inline issue!

**Review by RichKummer — APPROVED** [review]
LGTM! Thanks for the fixes @jkaeser 🙌

**Pull request merged**